qulice-checkstyle/src/main/resources/com/qulice/checkstyle/checks.xml

```diff
@@ -171,7 +171,7 @@
         <property name="message" value="Use 'java.nio.charset.StandardCharsets' instead"/>
     </module>
     <module name="RegexpSingleline">
-        <property name="format" value="^(?! *(/\*\*|\*|/\*|//)).*[\.\-\+%/\*&lt;&gt;] *$"/>
+        <property name="format" value="^(?! *(/\*\*|\*|/\*|//)).*[\.\-\+%/\*&lt;&gt;](?&lt;!(-&gt;)|&lt;*[A-Za-z]&gt;) *$"/>
```

Comment:
@paulodamaso  why a `*` after `<`?

Reply:
@krzyk Because it have to match `<T>` and `<String>` for example

Reply:
@paulodamaso Let me write the regexp here (just the added part):
```
(?<!(->)|<*[A-Za-z]>)
```
This will allow `<<<<<T>` but it won't allow `<Foo>`, you need to move the `*` after the `[A-Za-z]`.

BTW. test for that would be great :)

Reply:
@krzyk I dont think so, test is checking with `<Integer>` and it is passing, take a look

Reply:
@krzyk But `<<<<T>` and `String>` won't even compile, so whe should not bother with them

Reply:
@paulodamaso well yeah, but e.g.:
```
List<String> test = new ArrayList<     String>();
List<String> test2 = new ArrayList<
String>();
```
will compile.

And why have a broken regexp? :)

Instead of:
```
(?<!(->)|<*[A-Za-z]>)
```

it should be:
```
(?<!(->)|<[A-Za-z]*>)
```

Reply:
@krzyk Looks like the regex is right:
```
Caused by: java.util.regex.PatternSyntaxException: Look-behind group does not have an obvious maximum length near index 59
^(?! *(/\*\*|\*|/\*|//)).*[\.\-\+%/\*<>](?<!(->)|<[A-Za-z]*>) *$
                                                           ^
```

Reply:
@paulodamaso strange that it doesn't complain about maximum lenght in other case, ok let's merge this